packages/app/src/sandbox/compile.ts

```diff
@@ -377,31 +377,39 @@ function getDocumentHeight() {
   const { body } = document;
   const html = document.documentElement;
 
-  return Math.max(
-    body.scrollHeight,
-    body.offsetHeight,
-    html.clientHeight,
-    html.scrollHeight,
-    html.offsetHeight
-  );
+  return Math.max(body.scrollHeight, body.offsetHeight, html.offsetHeight);
 }
 
 function sendResize() {
   const height = getDocumentHeight();
 
   if (lastHeight !== height) {
-    if (document.body) {
-      dispatch({ type: 'resize', height });
-    }
+    dispatch({ type: 'resize', height });
   }
 
   lastHeight = height;
 }
 
-function initializeResizeListener() {
-  setInterval(sendResize, 5000);
+function initializeDOMMutationListener() {
+  if (typeof window === 'undefined') {
+    return;
+  }
+
+  // Listen on document body for any change that could trigger a resize of the content
+  // When a change is found, the sendResize function will determine if a message is dispatched
+  const observer = new MutationObserver(sendResize);
```

Comment:
Is MutationObserver available in all browsers? I'm thinking maybe an error can be thrown here.

Reply:
I will do this behind a check, since it's not a critical feature, but according to MDN it is supported everywhere